plugins/modules/podman_login.py

```diff
@@ -163,10 +179,23 @@ def main():
     password = module.params['password']
     certdir = module.params['certdir']
     tlsverify = module.params['tlsverify']
+    secret = module.params['secret']
     executable = module.get_bin_path(module.params['executable'], required=True)
 
+    podman_version = get_podman_version(module, fail=False)
+
+    if (
+        (podman_version is not None) and
+        (LooseVersion(podman_version) < LooseVersion('4.7.0')) and
+        secret
+    ):
+        module.fail_json(msg="secret option may not be used with podman < 4.7.0")
+
+    if username and (not secret) and (not password):
+        module.fail_json(msg="must specify password or secret with username")
```

Comment:
I am not sure how to model the username dependency, since it's:
Password requires username
Password conflicts with secret
Authfile conflicts with password and secret
Secret can appear without either password or username

Reply:
I see, in this case maybe better to use as:
```yaml
required_if=(
  ['password', 'username'],
),
mutually_exclusive=(
  ['password', 'secret'],
  ["authfile", "password"],
  ["authfile", "secret"],
),

```

Reply:
OK. I'm doing some more testing with authfile to make sure that all my assumptions are valid. What happens is that authfile is used if there's something in it already, but there's no guarantee that it will be. (Authfile is simply where podman-login writes the authentication stuff, so it's not really exclusive of username/password/secret. I'm working on amending the PR to reflect all of this, and I will make sure it has the DCO state with the next push.

Reply:
I've rebased all the existing commits onto tone branch. One issue I see with the current state is that passing username by itself (without either password or secret) causes an exception. I see how to make a password require a username, but I don't see a way to handle just a username using the existing mechanisms, so I'm re-instituting a version of that test in python.